go/mysql/auth_server_vault.go

```diff
@@ -140,15 +136,66 @@ func newAuthServerVault(addr string, timeout time.Duration, caCertPath string, p
 		vaultClient: client,
 		vaultPath:   path,
 		vaultTTL:    ttl,
-		method:      MysqlNativePassword,
 		entries:     make(map[string][]*AuthServerStaticEntry),
 	}
 
+	authMethodNative := NewMysqlNativeAuthMethod(a, a)
+	a.methods = []AuthMethod{authMethodNative}
+
 	a.reloadVault()
 	a.installSignalHandlers()
 	return a, nil
 }
 
+// AuthMethods returns the list of registered auth methods
+// implemented by this auth server.
+func (a *AuthServerVault) AuthMethods() []AuthMethod {
+	return a.methods
+}
+
+// DefaultAuthMethodDescription returns MysqlNativePassword as the default
+// authentication method for the auth server implementation.
+func (a *AuthServerVault) DefaultAuthMethodDescription() AuthMethodDescription {
+	return MysqlNativePassword
+}
+
+// HandleUser is part of the Validator interface. We
+// handle any user here since we don't check up front.
+func (a *AuthServerVault) HandleUser(user string) bool {
+	return true
+}
+
+// UserEntryWithHash is called when mysql_native_password is used.
+func (a *AuthServerVault) UserEntryWithHash(userCerts []*x509.Certificate, salt []byte, user string, authResponse []byte, remoteAddr net.Addr) (Getter, error) {
+	a.mu.Lock()
+	userEntries, ok := a.entries[user]
+	a.mu.Unlock()
+
+	if !ok {
+		return &StaticUserData{}, NewSQLError(ERAccessDeniedError, SSAccessDeniedError, "Access denied for user '%v'", user)
```

Comment:
nit: can return nil instead of `&StaticUserData{}`. Is there a reason to still return other than nil, this is also at other places as well.

Reply:
@harshit-gangal I copied this from how it was already working and assumed that it would need an object to be returned and not `nil`.